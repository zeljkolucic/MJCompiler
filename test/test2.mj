program MJProgram
	int x; int y; bool flag; const int one = 1; const int two = 2;
{
	bool main() 
	{
		x = 0;
		flag = false;
		y = x * flag;
		
		do {
			flag = flag + 2;
			x++;
		} while(x < 5);
		
		if(x == false)
			return 5;
		else if(flag == 2)
			flag++; 
			
		x = y + flag;
		one = 3;
		
		return 3;
	}
}