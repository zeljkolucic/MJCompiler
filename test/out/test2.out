Linija [6]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [7]: Pristup globalnoj promenljivoj flag.
DesignatorIdent(
 flag
) [DesignatorIdent]
Linija [8]: Pristup globalnoj promenljivoj y.
DesignatorIdent(
 y
) [DesignatorIdent]
Linija [8]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [8]: Pristup globalnoj promenljivoj flag.
DesignatorIdent(
 flag
) [DesignatorIdent]
Linija [11]: Pristup globalnoj promenljivoj flag.
DesignatorIdent(
 flag
) [DesignatorIdent]
Linija [11]: Pristup globalnoj promenljivoj flag.
DesignatorIdent(
 flag
) [DesignatorIdent]
Linija [12]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [13]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [15]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [17]: Pristup globalnoj promenljivoj flag.
DesignatorIdent(
 flag
) [DesignatorIdent]
Linija [18]: Pristup globalnoj promenljivoj flag.
DesignatorIdent(
 flag
) [DesignatorIdent]
Linija [20]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [20]: Pristup globalnoj promenljivoj y.
DesignatorIdent(
 y
) [DesignatorIdent]
Linija [20]: Pristup globalnoj promenljivoj flag.
DesignatorIdent(
 flag
) [DesignatorIdent]
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 
Prog MJProgram: notype, 0, 1 
   Var x: int, 0, 0 
   Var y: int, 1, 0 
   Var flag: , 2, 0 
   Con one: int, 1, 0 
   Con two: int, 2, 0 
   Meth main: , 0, 0 



